diff -Nuar iproute2-3.1.0.orig/Makefile iproute2-3.1.0/Makefile
--- iproute2-3.1.0.orig/Makefile	2011-11-18 08:53:50.000000000 +0800
+++ iproute2-3.1.0/Makefile	2013-01-06 14:28:30.000000000 +0800
@@ -1,3 +1,4 @@
+# Modifications were made by Cisco Systems, Inc. on or before Wed Nov 30 13:28:05 PST 2012
 DESTDIR=/usr/
 ROOTDIR=$(DESTDIR)
 LIBDIR=/usr/lib/
@@ -45,6 +46,7 @@
 
 Config:
 	sh configure $(KERNEL_INCLUDE)
+	sed 's/IP_CONFIG/#IP_CONFIG/g' <Config >Config
 
 install: all
 	install -m 0755 -d $(DESTDIR)$(SBINDIR)
