diff -Naur linux-3.4.103.orig/arch/arm/mach-msm/board-ipq806x.c linux-3.4.103/arch/arm/mach-msm/board-ipq806x.c
--- linux-3.4.103.orig/arch/arm/mach-msm/board-ipq806x.c	2015-03-31 14:16:52.058771613 -0700
+++ linux-3.4.103/arch/arm/mach-msm/board-ipq806x.c	2015-04-01 08:26:21.309474137 -0700
@@ -1,3 +1,4 @@
+/* Modifications were made by Linksys on or before Wed Apr  1 08:24:56 PDT 2015 */
 /* * Copyright (c) 2012-2014 The Linux Foundation. All rights reserved.* */
 /* Copyright (c) 2011-2012, The Linux Foundation. All rights reserved.
  *
@@ -1656,7 +1657,8 @@
 		rst[2] = -1;
 		pwr[2] = -1;
 		no_vreg[2] = 1;
-		msm_pcie_platform_data[1].force_gen1 = 1;
+	  /* patch from QCA 04-01-2015 c/o mtien@qca.qualcomm.com */
+		msm_pcie_platform_data[0].force_gen1 = 1;
 	}
 
 	for (i = 0; i < CONFIG_MSM_NUM_PCIE; i++) {
