diff -rcN iptables-1.4.4/include/linux/netfilter_ipv4/ipt_dscp.h iptables-1.4.4.new/include/linux/netfilter_ipv4/ipt_dscp.h
*** iptables-1.4.4/include/linux/netfilter_ipv4/ipt_dscp.h	1969-12-31 16:00:00.000000000 -0800
--- iptables-1.4.4.new/include/linux/netfilter_ipv4/ipt_dscp.h	2012-02-28 09:10:34.483299781 -0800
***************
*** 0 ****
--- 1,22 ----
+ /* Modifications were made by Cisco Systems, Inc. on or before Fri Mar 23 16:21:07 PDT 2012 */
+ /* iptables module for matching the IPv4 DSCP field
+  *
+  * (C) 2002 Harald Welte <laforge@gnumonks.org>
+  * This software is distributed under GNU GPL v2, 1991
+  * 
+  * See RFC2474 for a description of the DSCP field within the IP Header.
+  *
+  * ipt_dscp.h,v 1.3 2002/08/05 19:00:21 laforge Exp
+ */
+ #ifndef _IPT_DSCP_H
+ #define _IPT_DSCP_H
+ 
+ #include <linux/netfilter/xt_dscp.h>
+ 
+ #define IPT_DSCP_MASK	XT_DSCP_MASK
+ #define IPT_DSCP_SHIFT	XT_DSCP_SHIFT
+ #define IPT_DSCP_MAX	XT_DSCP_MAX
+ 
+ #define ipt_dscp_info	xt_dscp_info
+ 
+ #endif /* _IPT_DSCP_H */
diff -rcN iptables-1.4.4/include/linux/netfilter_ipv4/ipt_DSCP.h iptables-1.4.4.new/include/linux/netfilter_ipv4/ipt_DSCP.h
*** iptables-1.4.4/include/linux/netfilter_ipv4/ipt_DSCP.h	1969-12-31 16:00:00.000000000 -0800
--- iptables-1.4.4.new/include/linux/netfilter_ipv4/ipt_DSCP.h	2012-02-28 09:10:57.027329110 -0800
***************
*** 0 ****
--- 1,19 ----
+ /* Modifications were made by Cisco Systems, Inc. on or before Fri Mar 23 16:21:07 PDT 2012 */
+ /* iptables module for setting the IPv4 DSCP field
+  *
+  * (C) 2002 Harald Welte <laforge@gnumonks.org>
+  * based on ipt_FTOS.c (C) 2000 by Matthew G. Marsh <mgm@paktronix.com>
+  * This software is distributed under GNU GPL v2, 1991
+  * 
+  * See RFC2474 for a description of the DSCP field within the IP Header.
+  *
+  * ipt_DSCP.h,v 1.7 2002/03/14 12:03:13 laforge Exp
+ */
+ #ifndef _IPT_DSCP_TARGET_H
+ #define _IPT_DSCP_TARGET_H
+ #include <linux/netfilter_ipv4/ipt_dscp.h>
+ #include <linux/netfilter/xt_DSCP.h>
+ 
+ #define ipt_DSCP_info xt_DSCP_info
+ 
+ #endif /* _IPT_DSCP_TARGET_H */
