--- old/pppd/lcp.c	2011-08-05 14:44:53.000000000 -0700
+++ new/pppd/lcp.c	2011-08-05 14:42:47.000000000 -0700
@@ -1499,6 +1499,7 @@
     lcp_options *go = &lcp_gotoptions[f->unit];
     lcp_options *ho = &lcp_hisoptions[f->unit];
     lcp_options *ao = &lcp_allowoptions[f->unit];
+    lcp_options *wo = &lcp_wantoptions[f->unit];
     u_char *cip, *next;		/* Pointer to current and next CIs */
     int cilen, citype, cichar;	/* Parsed len, type, char value */
     u_short cishort;		/* Parsed short value */
@@ -1553,7 +1554,12 @@
 	     * No need to check a maximum.  If he sends a large number,
 	     * we'll just ignore it.
 	     */
-	    if (cishort < MINMRU) {
+	    // if (cishort < MINMRU) {
+            /*
+             * cshen@cisco.com 08/05/2011
+             *   We don't want to ignor larger than our configured mru as well.
+             */
+	    if (cishort < MINMRU || cishort > wo->mru) {
 		orc = CONFNAK;		/* Nak CI */
 		PUTCHAR(CI_MRU, nakp);
 		PUTCHAR(CILEN_SHORT, nakp);
