diff -Naur attr-2.4.46.orig/include/config.h.in attr-2.4.46/include/config.h.in
--- attr-2.4.46.orig/include/config.h.in	2010-01-22 14:55:11.000000000 -0800
+++ attr-2.4.46/include/config.h.in	2013-12-05 08:58:46.265612203 -0800
@@ -1,3 +1,4 @@
+/* Modifications were made by Linksys on or before Thu Dec  5 08:58:46 PST 2013 */
 /*
  * Copyright (c) 2002 Silicon Graphics, Inc.  All Rights Reserved.
  *
@@ -29,14 +30,14 @@
    */
 #undef HAVE_ALLOCA_H
 
-#ifdef ENABLE_GETTEXT
+/*#ifdef ENABLE_GETTEXT
 # include <libintl.h>
 # define _(x)			gettext(x)
-#else
+#else*/
 # define _(x)			(x)
 # define textdomain(d)		do { } while (0)
 # define bindtextdomain(d,dir)	do { } while (0)
-#endif
+/*#endif*/
 #include <locale.h>
 
 #endif	/* __CONFIG_H__ */
