diff -Naur igmpproxy-0.1.orig/src/igmpproxy.c igmpproxy-0.1/src/igmpproxy.c
--- igmpproxy-0.1.orig/src/igmpproxy.c	2009-08-27 12:32:28.000000000 -0700
+++ igmpproxy-0.1/src/igmpproxy.c	2010-01-06 20:45:26.000000000 -0800
@@ -72,7 +72,8 @@
 int main( int ArgCn, char *ArgVc[] ) {
 
     // Parse the commandline options and setup basic settings..
-    for (int c; (c = getopt(ArgCn, ArgVc, "vdh")) != -1;) {
+    int c; // Added build patch, Dave Tuttle, Cisco Systems, Inc.,2010-01-06
+    for (; (c = getopt(ArgCn, ArgVc, "vdh")) != -1;) {
         switch (c) {
         case 'd':
             Log2Stderr = true;
