diff -Naur igmpproxy-0.1.orig/src/igmpproxy.c igmpproxy-0.1/src/igmpproxy.c
--- igmpproxy-0.1.orig/src/igmpproxy.c  2009-08-27 12:32:28.000000000 -0700
+++ igmpproxy-0.1/src/igmpproxy.c       2010-10-12 14:22:26.000000000 -0700
@@ -37,6 +37,7 @@
 */
 
 #include "igmpproxy.h"
+#include <sys/sysinfo.h>
 
 static const char Usage[] = 
 "Usage: igmpproxy [-h] [-d] [-v [-v]] <configfile>\n"
@@ -221,6 +222,16 @@
 
 }
 
+static void getuptime(struct timeval *tv)
+{   // Fixed CPU usage issue, Sridhar Ramaswamy, Cisco Systems, Inc.,2010-10-12
+    struct sysinfo si;
+
+    sysinfo( &si );
+
+    tv->tv_sec = si.uptime;
+    tv->tv_usec = 0;
+}
+
 /**
 *   Main daemon loop.
 */
@@ -238,7 +249,7 @@
 
     // Initialize timer vars
     difftime.tv_usec = 0;
-    gettimeofday(&curtime, NULL);
+    getuptime(&curtime);
     lasttime = curtime;
 
     // First thing we send a membership query in downstream VIF's...
@@ -301,14 +312,14 @@
             /*
              * If the select timed out, then there's no other
              * activity to account for and we don't need to
-             * call gettimeofday.
+             * call getuptime.
              */
             if (Rt == 0) {
                 curtime.tv_sec = lasttime.tv_sec + secs;
                 curtime.tv_usec = lasttime.tv_usec;
                 Rt = -1; /* don't do this next time through the loop */
             } else {
-                gettimeofday(&curtime, NULL);
+                getuptime(&curtime); // Fixed CPU usage issue, Sridhar Ramaswamy, Cisco Systems, Inc.,2010-10-12
             }
             difftime.tv_sec = curtime.tv_sec - lasttime.tv_sec;
             difftime.tv_usec += curtime.tv_usec - lasttime.tv_usec;

